manager/src/main/java/com/redhat/service/smartevents/manager/BridgesServiceImpl.java

```diff
@@ -90,6 +91,53 @@ public Bridge createBridge(String customerId, String organisationId, String owne
         return bridge;
     }
 
+    @Override
+    @Transactional
+    public Bridge updateBridge(String bridgeId, String customerId, BridgeRequest bridgeRequest) {
+        // Extract existing definition
+        Bridge existingBridge = getBridge(bridgeId, customerId);
+
+        if (!existingBridge.isActionable()) {
+            throw new BridgeLifecycleException(String.format("Bridge with id '%s' for customer '%s' is not in an actionable state.",
+                    bridgeId,
+                    customerId));
+        }
+
+        BridgeDefinition existingDefinition = existingBridge.getDefinition();
+
+        // Validate update.
+        // Name cannot be updated.
+        if (!Objects.equals(existingBridge.getName(), bridgeRequest.getName())) {
+            throw new BadRequestException("It is not possible to update the Bridge's name.");
+        }
+
+        // Construct updated definition
+        Action updatedErrorHandler = bridgeRequest.getErrorHandler();
+        BridgeDefinition updatedDefinition = new BridgeDefinition(Objects.nonNull(updatedErrorHandler) ? updatedErrorHandler : null);
+
+        // No need to update CRD if the definition is unchanged
+        if (existingDefinition.equals(updatedDefinition)) {
+            return existingBridge;
+        }
+
+        // Create new definition copying existing properties
+        existingBridge.setModifiedAt(ZonedDateTime.now());
+        existingBridge.setStatus(ManagedResourceStatus.ACCEPTED);
+        existingBridge.setDependencyStatus(ManagedResourceStatus.ACCEPTED);
+        existingBridge.setDefinition(updatedDefinition);
+        existingBridge.setGeneration(existingBridge.getGeneration() + 1);
+
+        // Bridge and Work should always be created in the same transaction
+        workManager.schedule(existingBridge);
```

Comment:
AFAIK the work handling should be synchronized with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/920 (adjusting the PR once the other one is merged)

Reply:
Correct. I would suggest #920 be merged first.

The _approach_ to editing Bridges is still not agreed across the team.

I _thought_ we was going to have agreement on Friday but it never came to be.

I'll move this back to draft.